src/Entity/TaskListRepository.php

```diff
@@ -0,0 +1,184 @@
+                'org.name AS organizationName',
+                // objects are listed below to force them being hydrated / pre-fetched by Doctrine
+                // https://www.doctrine-project.org/projects/doctrine-orm/en/3.2/reference/dql-doctrine-query-language.html#result-format
+                'taskPackage',
+                'package',
+                'incidents',
+            ])
+            ->from(Task::class, 't')
+            ->leftJoin('t.delivery', 'd')
+            ->leftJoin('d.order', 'o')
+            ->leftJoin('t.organization', 'org')
+            ->leftJoin('t.packages', 'taskPackage')
+            ->leftJoin('taskPackage.package', 'package')
+            ->leftJoin('t.incidents', 'incidents')
+            ->where('t.id IN (:taskIds)')
+            ->andWhere('t.status != :statusCancelled')
+            ->setParameter('taskIds', $orderedTaskIds) // using IN might cause problems with large number of tasks
+            ->setParameter('statusCancelled', Task::STATUS_CANCELLED)
+            ->getQuery()
+            ->getResult();
+
+        $tasksByDeliveryId = array_reduce($tasksQueryResult, function ($carry, $row) {
+            $deliveryId = $row['deliveryId'] ?? null;
+
+            if (null === $deliveryId) {
+                return $carry;
+            }
+
+            $task = $row[0];
+            $carry[$deliveryId][] = $task;
+            return $carry;
+        }, []);
+
+        $tasks = array_map(function ($row) use ($tasksByDeliveryId) {
+            $task = $row[0];
+            $deliveryId = $row['deliveryId'] ?? null;
+
+            $taskPackages = [];
+            $weight = null;
+
+            $tasksInTheSameDelivery = $deliveryId ? $tasksByDeliveryId[$deliveryId] : [];
+
+            if ($task->isPickup()) {
+                // for a pickup in a delivery, the serialized weight is the sum of the dropoff weight and
+                // the packages are the "sum" of the dropoffs packages
+                foreach ($tasksInTheSameDelivery as $task) {
+                    if ($task->isPickup()) {
+                        continue;
+                    }
+
+                    $taskPackages = array_merge($taskPackages, $task->getPackages()->toArray());
+                    $weight += $task->getWeight();
+                }
+            } else {
+                $taskPackages = $task->getPackages()->toArray();
+                $weight = $task->getWeight();
+            }
+
+            $task = $row[0];
+            $taskDto = new MyTaskDto(
+                $task->getId(),
+                $task->getCreatedAt(),
+                $task->getUpdatedAt(),
+                $task->getType(),
+                $task->getStatus(),
+                $task->getAddress(),
+                $task->getDoneAfter(),
+                $task->getDoneBefore(),
+                $task->getPrevious()?->getId(),
+                $task->getNext()?->getId(),
+                $task->getTags(),
+                $task->isDoorstep(),
```

Comment:
for your information isDoorstep is soon to be deprecated (i removed it in an other branch)